### PR TITLE
Run batching queries in parallel (#176)

### LINKED ISSUE #176: batching implementations should (optionally?) run queries in parallel
Currently they use `await` to yield before running the next query.

### PR DESCRIPTION
**Pull request by DxCx**
TODO:

- [x] Update CHANGELOG.md with your change (include reference to issue & this PR)
- [X] Make sure all of the significant new logic is covered by tests
- [ ] Rebase your changes on master so that they can be merged easily
- [X] Make sure all tests and linter rules pass

Closes #176

### CODE REVIEW

**Comment by DxCx** [comment]
@helfer Please approve, i'll rebase & merge once the other PRs are merged..

File: packages/graphql-server-express/src/apolloServerHttp.test.ts [hunk 1]
```diff
@@ -154,7 +154,9 @@ const version = 'modern';
 describe(`GraphQL-HTTP (apolloServer) tests for ${version} express`, () => {
   describe('POST functionality', () => {
 
-    it('allows gzipped POST bodies', async () => {
+    it('allows gzipped POST bodies', async function () {
```
Comment (helfer) [inline packages/graphql-server-express/src/apolloServerHttp.test.ts]: How is this change related to this PR?

Comment (DxCx) [inline packages/graphql-server-express/src/apolloServerHttp.test.ts]: Node 4 is much slower then Node 6.
I increase those 2 tests timeout since there is timeout after exhausting node abit..

File: packages/graphql-server-integration-testsuite/src/index.ts [hunk 2]
```diff
@@ -457,6 +469,29 @@ export default (createApp: CreateAppFunc, destroyApp?: DestroyAppFunc) => {
           });
       });
 
+       it('can handle batch requests in parallel', function() {
```
Comment (glasser) [inline packages/graphql-server-integration-testsuite/src/index.ts]: Another potential testing strategy that doesn't require the test to take a long time is to have much shorter timeouts in the queries but to expect the part of one query after a timeout to be able to observe the effect of another query.

Comment (DxCx) [inline packages/graphql-server-integration-testsuite/src/index.ts]: yeah i thought about creating a promise and resolve it on the other query.
but it felt just too bad..
this is probably what i would've done if it was not going through the whole query engine and such.
